src/hooks/core/push-notification/usePushNotificationHook.ts

```diff
@@ -0,0 +1,218 @@
+import PushNotification from 'react-native-push-notification';
+import PushNotificationIOS from '@react-native-community/push-notification-ios';
+import messaging from '@react-native-firebase/messaging';
+import React, {useState} from 'react';
+import {Platform} from 'react-native';
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Consider using a utility function to determine the platform. This will make it easier to add more platforms in the future.
```diff
- const isIos = Platform.OS === 'ios';
+ const isIos = isPlatformIOS();
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->